utilities/download_old_firefox.py

```diff
@@ -7,45 +7,19 @@
 if __name__ == "__main__":
     base_url = "https://download-installer.cdn.mozilla.net"
     today = date.today()
-    # Set correct month for URL builder
-    if len(f"{today.month}") < 2:
-        current_month = f"0{today.month}"
-    else:
-        current_month = f"{today.month}"
-
-    # Set correct month for URL builder
-    if len(f"{today.day}") < 2 and int(today.day - 5) < 0:
-        download_day = "01"
-    elif int(today.day) < 15:
-        download_day = f"0{today.day - 5}"
-    else:
-        download_day = f"{today.day - 5}"
-
-    download_month = f"{current_month}"
-
-    # if its a new month just grab the 28th day build of last month and build URL
-    if int(download_day) < 6:
-        if int(current_month) < 10:
-            download_month = f"0{int(current_month) - 1}"
-        else:
-             download_month = f"{int(current_month) - 1}"
-        download_dir = f"{base_url}/pub/firefox/nightly/{today.year}/{download_month}/"
-        html = requests.get(download_dir)
-
-        soup = BeautifulSoup(html.text, "html.parser")
-        page_link = soup.find_all(
-            href=re.compile(f"{today.year}-{download_month}-28.*-mozilla-central")
-        )
-    else:
-        download_dir = f"{base_url}/pub/firefox/nightly/{today.year}/{download_month}/"
-        html = requests.get(download_dir)
-
-        soup = BeautifulSoup(html.text, "html.parser")
-        page_link = soup.find_all(
-            href=re.compile(
-                f"{today.year}-{download_month}-{download_day}.*-mozilla-central"
-            )
+    download_date = today - timedelta(weeks=4)
+    download_date = download_date.replace(day=15)
```

Comment:
R+ but I was thinking a bit more about this last night and realized there is still [at least] one edge case we may or may not care about... For months with more than 4 weeks/28 days, the math gets a bit odd and you get back the same month if you're on the last 1-3 days of the month.

```py
from datetime import date, timedelta

base_url = "https://download-installer.cdn.mozilla.net"

for i in range(3, 12):
    today = date.fromisoformat(f"2019-{i:02}-30")
    firefox_date = today - timedelta(weeks=4)
    firefox_date2 = firefox_date.replace(day=15)
    print(f"today={today}; firefox_date={firefox_date}; firefox_date2={firefox_date2}")
```

So 2019-03-30 minus 4 weeks/28 days is 2019-03-02, then resetting the day to the 15th of the month results in a 2 week old build vs a 6 week old build.
```
today=2019-03-30; firefox_date=2019-03-02; firefox_date2=2019-03-15
today=2019-04-30; firefox_date=2019-04-02; firefox_date2=2019-04-15
today=2019-05-30; firefox_date=2019-05-02; firefox_date2=2019-05-15
today=2019-06-30; firefox_date=2019-06-02; firefox_date2=2019-06-15
today=2019-07-30; firefox_date=2019-07-02; firefox_date2=2019-07-15
today=2019-08-30; firefox_date=2019-08-02; firefox_date2=2019-08-15
today=2019-09-30; firefox_date=2019-09-02; firefox_date2=2019-09-15
today=2019-10-30; firefox_date=2019-10-02; firefox_date2=2019-10-15
today=2019-11-30; firefox_date=2019-11-02; firefox_date2=2019-11-15
```

Reply:
That's fine. I just chose 15 randomly, as long as it's a few days old, it will trigger the update message on startup, that's all I need to happen.